app/(app)/speakers.tsx

```diff
@@ -1,44 +1,69 @@
-import { Link, Stack } from 'expo-router';
-import React from 'react';
-import StyledText from '../../components/common/StyledText';
+import { AntDesign } from '@expo/vector-icons';
+import { useTheme } from '@react-navigation/native';
+import { Stack, useRouter } from 'expo-router';
+import React, { useEffect, useState } from 'react';
+import { FlatList, StyleSheet, View } from 'react-native';
+import SpeakerCard from '../../components/cards/SpeakerCard';
+import Space from '../../components/common/Space';
 import MainContainer from '../../components/container/MainContainer';
+import type { Session, Speaker } from '../../global';
+import { Sessions } from '../../mock/sessions';
+import { Speakers } from '../../mock/speakers';
 
-// TODO: implement speakers page
-/**
- * TASK:
- * - Should display a grid view of speakers
- * - <SpeakerCard /> should be used to display each speaker
- */
-// TODO: this is dummy data, replace with real data from mock/speakers.ts
-const _speakers = [
-  {
-    id: '1',
-    name: 'John Doe',
-  },
-  {
-    id: '2',
-    name: 'Jane Doe',
-  },
-];
+interface SpeakerItem extends Speaker {
+  sessions: Array<Session>;
+}
+
+const SpeakersPage = () => {
+  const { colors } = useTheme();
+  const router = useRouter();
+  const [data, setData] = useState<Array<SpeakerItem>>([]);
+
+  const getSpeakerSessions = () => {
+    const speakerArray = [...Speakers.data] as Array<SpeakerItem>;
+
+    speakerArray.map((speaker) => {
+      speaker.sessions = Sessions.data.filter((session) =>
+        session.speakers.some((_speaker) => _speaker.name === speaker.name),
+      );
+    });
+    setData(speakerArray);
+  };
+
+  useEffect(() => {
+    getSpeakerSessions();
+  }, []);
```

Comment:
We do not need to use an Effect here, we can move the function outside the component and use its result straight away:

```tsx
const getSpeakerSessions = () => {
  const speakerArray = [...Speakers.data] as Array<SpeakerItem>;

  speakerArray.map((speaker) => {
    speaker.sessions = Sessions.data.filter((session) =>
      session.speakers.some((_speaker) => _speaker.name === speaker.name),
    );
  });
  return speakerArray;
};

const SpeakersPage = () => {
  const data = getSpeakerSessions();
  //<rest of the component>
```